twilio/provider.go

```diff
@@ -18,14 +18,14 @@ func Provider() *schema.Provider {
 		Schema: map[string]*schema.Schema{
 			"account_sid": {
 				Type:        schema.TypeString,
-				DefaultFunc: schema.EnvDefaultFunc("TWILIO_ACCOUNT_SID", nil),
-				Description: "Your Account SID can be found on the Twilio dashboard at www.twilio.com/console.",
+				DefaultFunc: schema.MultiEnvDefaultFunc([]string{"TWILIO_ACCOUNT_SID", "TWILIO_API_KEY"}, nil),
```

Comment:
This doesn't support using the 3 env vars concurrently. I.e., use `TWILIO_ACCOUNT_SID` for v2010 APIs with auth coming from `TWILIO_API_KEY` and `TWILIO_API_SECRET`.

Reply:
I think as long as there is a fallback for the TWILIO_SUBACCOUNT_SID to TWILIO_ACCOUNT_SID it should work